packages/twenty-front/src/modules/settings/components/SettingsOptionCardContent.tsx

```diff
@@ -52,44 +94,99 @@ const StyledToggle = styled(Toggle)`
   margin-left: auto;
 `;
 
+const StyledSelect = styled(Select)`
+  margin-left: auto;
+  width: 120px;
+`;
+
 const StyledCover = styled.span`
   cursor: pointer;
   inset: 0;
   position: absolute;
 `;
 
-export const SettingsOptionCardContent = ({
-  Icon,
-  title,
-  description,
-  divider,
-  checked,
-  onChange,
-}: SettingsOptionCardContentProps) => {
+export const SettingsOptionCardContent = (
+  props: SettingsOptionCardContentProps,
+) => {
   const theme = useTheme();
-
   const toggleId = useId();
+  const {
+    Icon,
+    title,
+    description,
+    divider,
+    disabled = false,
+    advancedMode = false,
+  } = props;
+
+  const renderControl = () => {
+    if (props.variant === 'toggle') {
+      return (
+        <StyledToggle
+          id={toggleId}
+          value={props.checked}
+          onChange={props.onChange}
+          disabled={disabled}
+          color={advancedMode ? theme.color.yellow : theme.color.blue}
+        />
+      );
+    }
+
+    if (props.variant === 'select') {
+      return (
+        <StyledSelect
+          className={props.selectClassName}
+          dropdownWidth={props.fullWidth ? 'auto' : 120}
+          disabled={disabled}
+          dropdownId={props.dropdownId}
+          value={props.value}
+          onChange={props.onChange}
+          options={props.options}
+          variant="small"
+        />
+      );
+    }
+
+    return (
+      <SettingsCounter
+        value={props.value}
+        onChange={props.onChange}
+        minValue={props.minValue}
+        maxValue={props.maxValue}
+      />
+    );
+  };
+
+  const getDescription = () => {
+    if (props.variant === 'counter') {
```

Comment:
it's a not good to use functions to compute variables, this will create a new function on each re-render I believe

Reply:
will be simplified by having different components